src/commands/deck.ts

```diff
@@ -340,12 +368,74 @@ export class DeckCommand extends Command {
 		const outFile = typedDeckToYdk(deck);
 
 		const content = await this.generateProfile(deck, resultLanguage, !isStacked, outUrl);
+
+		useLocale(resultLanguage);
+		// prepare interaction button for FTP upload
+		const row = new ActionRowBuilder<ButtonBuilder>().addComponents(
+			new ButtonBuilder()
+				.setCustomId("ftp")
+				.setLabel(t`Upload to YGOPRODECK`)
+				.setStyle(ButtonStyle.Primary)
+		);
+
+		// a string is interpreted as a path, to upload it as a file we need a Buffer
+		// we might also want it again for the FTP upload
+		const deckBuffer = Buffer.from(outFile, "utf-8");
+
+		// save to re-use when we later edit the message
+		const embeds = addNotice(content);
+		const attachment = new AttachmentBuilder(deckBuffer).setName("deck.ydk");
+
 		const end = Date.now();
-		await interaction.editReply({
-			embeds: addNotice(content),
-			// a string is interpreted as a path, to upload it as a file we need a Buffer
-			files: [new AttachmentBuilder(Buffer.from(outFile, "utf-8")).setName("deck.ydk")]
+		const response = await interaction.editReply({
+			embeds,
+			files: [attachment],
+			components: [row]
 		});
+
+		const filter = (i: ButtonInteraction): boolean => {
+			// only allow the OP to click
+			return i.user.id === interaction.user.id;
+		};
+
+		// TODO: determine ideal timeout length.
+		// we don't await this promise, we set up the callback and then let the method complete
+		response
+			.awaitMessageComponent({ filter, componentType: ComponentType.Button, time: 60000 })
+			.then(async i => {
+				await i.deferReply({ ephemeral: !isPublic });
+
+				// upload deck with FTP
+				const filename = `${i.id}.ydk`; // use interaction ID for unique filename
+				await this.upload(filename, deckBuffer);
```

Comment:
For specifically error handling for FTP, you can wrap this in a try catch and then send an appropriate reply and remove the button.